### PR TITLE
Process attestation: reduce checkpoint copies

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Whenever the node processes a single attestation, it automatically copy checkpoint twice. This is unnecessary and inefficient as the number of attestation in block grows. This PR implemented two functions `MatchCurrentJustifiedCheckpoint` and `MatchPreviousJustifiedCheckpoint` to enable beacon state checkpoint comparisons without copies. 

Also minor clean up on retrieving previous and current epochs and removing unnecessary `data.Target.Epoch != ffgTargetEpoch` check. (not in the spec)

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

N/A

### CODE REVIEW

File: beacon-chain/state/getters.go [hunk 1]
```diff
@@ -1016,6 +1017,44 @@ func (b *BeaconState) currentJustifiedCheckpoint() *ethpb.Checkpoint {
...
+// the current justified checkpoint in state.
+func (b *BeaconState) MatchCurrentJustifiedCheckpoint(c *ethpb.Checkpoint) bool {
+	if !b.HasInnerState() {
+		return false
+	}
+	if b.state.CurrentJustifiedCheckpoint == nil {
+		return false
+	}
+
+	b.lock.RLock()
```
Comment (rauljordan) [inline beacon-chain/state/getters.go:1030]: Why do we need locks here? We are just accessing struct fields for reading, there shouldn't be any contention across goroutines

**Comment by rauljordan** [comment]
Just a small comment on use of locks and a small conflict

File: beacon-chain/state/getters.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/state/getters.go:1030]: Ah i see. Removing...

**Comment by terencechain** [comment]
> Just a small comment on use of locks and a small conflict

Addressed. Thanks

**Review by rauljordan — APPROVED** [review]

**Pull request merged**